### ISSUE TITLE
Destruct fails when there are lets in the parameters of an inductive

### ISSUE THREAD

**Issue by SkySkimmer** [coq/coq#9466]
~~~coq
Inductive foo (x:=nat) : Type := C : foo.

Goal  foo -> True.
Proof.
  intros H. destruct H.
  (* Error: decompose_prod_n_assum: not enough assumptions *)
Abort.
~~~
cc @ppedrot

**Comment by ppedrot** [comment]
FTR I discovered that changing the representation of constructor types in the kernel, and I am about to submit a PR fixing it.

**Comment by SkySkimmer** [comment]
Fixed in 8.18
